test/unit/pybind/test_env_variables.py

```diff
@@ -0,0 +1,28 @@
+import nmodl.dsl as nmodl
+
+
+def test_visitors():
+    """
+    Make sure all of the env variables are set when running the NMODL Python package
+    """
+
+    mod_string = """
+    NEURON {
+        USEION na READ ena WRITE ina
+        RANGE gna
+    }
+    BREAKPOINT {
+        ina = gna*(v - ena)
+    }"""
+
+    # parse NMDOL file (supplied as a string) into AST
+    driver = nmodl.NmodlDriver()
+    AST = driver.parse_string(mod_string)
+    # run SymtabVisitor to generate Symbol Table
+    nmodl.symtab.SymtabVisitor().visit_program(AST)
+    # constant folding, inlining & local variable renaming passes
+    nmodl.visitor.ConstantFolderVisitor().visit_program(AST)
+    nmodl.visitor.InlineVisitor().visit_program(AST)
+    nmodl.visitor.LocalVarRenameVisitor().visit_program(AST)
+    # run CONDUCTANCE visitor
+    nmodl.visitor.SympyConductanceVisitor().visit_program(AST)
```

Comment:
Is this the first unit-test that runs NMODL as a module? If no why do the others not fail?

Reply:
I'm not 100% sure, but I think this has to do with the `BREAKPOINT` statement, as the following code demonstrates (tested on latest master):

```python
# example.py
import nmodl
from nmodl.dsl import visitor


driver = nmodl.NmodlDriver()
bad_string = """
NEURON {
    USEION na READ ena WRITE ina
    RANGE gna
}
BREAKPOINT {
    ina = gna*(v - ena)
}"""

good_string = """NEURON  {
    SUFFIX NaTs2_t
    RANGE mInf, hInf
}

ASSIGNED {
    mInf
    hInf
}

STATE   {
    m
    h
}

DERIVATIVE states   {
    m' = mInf-m
    h' = hInf-h
}"""

visitor.SympyConductanceVisitor().visit_program(driver.parse_string(good_string))
# call order w.r.t. above doesn't matter, below will always raise an error
visitor.SympyConductanceVisitor().visit_program(driver.parse_string(bad_string))
```
This seems to give:
```plaintext
[NMODL] [critical] :: NMODL_PYLIB environment variable must be set to load embedded python
Traceback (most recent call last):
  File "example.py", line 37, in <module>
    visitor.SympyConductanceVisitor().visit_program(driver.parse_string(bad_string))
RuntimeError: NMODL_PYLIB not set
```

I am guessing we are only calling sympy (and hence the embedded interpreter) when the parser processes `BREAKPOINT`.

Reply:
Thank you!